scripts/core/editor3/components/Editor3Component.tsx

```diff
@@ -78,8 +64,17 @@ export function isAllowedMediaType(mediaType: string, dataTransfer: DataTransfer
 */
 export function canDropMedia(e, editorConfig) {
     const {editorFormat, readOnly, singleLine} = editorConfig;
-    const isValidMedia = !!getValidMediaType(e.originalEvent);
     const supportsMedia = !readOnly && !singleLine && editorFormat.indexOf('media') !== -1;
+    const mediaType = getValidMediaType(e.originalEvent);
+    const dataTransfer = e.originalEvent.dataTransfer;
+    let isValidMedia = !!mediaType;
+
+    if (mediaType === 'Files' && dataTransfer.files) {
+        // checks if files dropped from external folder are valid or not
+        isValidMedia = Object.values(dataTransfer.files).every(
```

Comment:
If `isValidMedia` is false at line 74, the entire function will always return false. Can you define a local variable and do an early return instead of overwriting the variable in another block scope?

```
if (mediaType === 'Files' && dataTransfer.files) {
    const isValidFileType = Object.values(dataTransfer.files).every(...);

    if (!isValidFileType) {
        return false;
    }
}

```